File: app/src/main/java/com/donnfelker/kotlinmix/KotlinMixApplication.java
package com.donnfelker.kotlinmix;

import android.app.Application;

import io.realm.Realm;
import io.realm.RealmConfiguration;

public class KotlinMixApplication extends Application {
    @Override
    public void onCreate() {
        super.onCreate();

        RealmConfiguration config = new RealmConfiguration.Builder(this)
                .name("kotlinmix.realm")
                .deleteRealmIfMigrationNeeded()
                .build();

        Realm.setDefaultConfiguration(config);
    }
}

File: app/src/main/kotlin/com/donnfelker/kotlinmix/KotlinMixApplication.kt
package com.donnfelker.kotlinmix

import android.app.Application

import io.realm.Realm
import io.realm.RealmConfiguration

class KotlinMixApplication : Application() {
    override fun onCreate() {
        super.onCreate()

        Realm.init(this)

        val config = RealmConfiguration
                .Builder()
//                .deleteRealmIfMigrationNeeded()
                .name("kotlinmix.realm")
                .build()

        Realm.setDefaultConfiguration(config)
    }
}
